probe-rs/src/core/registers.rs

```diff
@@ -63,26 +64,30 @@ impl Display for RegisterRole {
 pub struct CoreRegister {
     /// The architecture specific name of the register. This may be identical, or similar to [`RegisterRole`].
     pub(crate) name: &'static str,
+    // /// Some architectures have multiple names for the same register, depending on the context and the role of the register.
+    // /// This field contains a list of aliases or synonyms for the register.
+    // pub(crate) aliases: Option<&'static [&'static str]>,
     pub(crate) id: RegisterId,
     /// If the register plays a special role during program execution and exception handling, this field will be set to the role.
     /// Otherwise, it will be `None` and the register is a general purpose register.
-    pub(crate) role: Option<RegisterRole>,
+    pub(crate) roles: Option<&'static [RegisterRole]>,
```

Comment:
Maybe it would be easier to just always have roles? Is there a useful difference between an empty slice and `None`?

Reply:
@Tiwalun, I've made the change, but now that I look at the design, an additional choice would be to remove the `name` field, and add a role, such as `RegisterRole::Base(&'static str)` to capture the equivalent. 

It should be an easy refactor ... what do you think?

Reply:
I think it would make it a bit cleaner, no special handling of `name` is needed anymore. But a separate name is fine as well.